united.py

```diff
@@ -28,10 +54,224 @@ def parse_date(created):
     """
     time = datetime.strptime(created[:-3], '%m/%d/%Y %I:%M:%S')
     hour = time.strftime('%H')
-    hour = int(hour) if created.endswith('AM') else int(hour)+12
+    hour = int(hour) if created.endswith('AM') else int(hour) + 12
     return datetime(time.year, time.month, time.day, hour, time.minute, 0)
 
 
+def is_nth_weekday(nth, daynum, year, month):  # start counting the daynum from monday = 0
+    return calendar.Calendar(nth).monthdatescalendar(
+        year,
+        month
+    )[daynum][0]
+
+
+def get_parent_object_node(node):
+    while node.parentNode:
+        node = node.parentNode
+        if node.nodeName == "Object":
+            return node
+
+
+def all_station_in_date_frame(created):
+
+    station_data_in_date = collection.getElementsByTagName('date_selected')
+    station_data_in_date.sort(key=lambda x: int(x.attributes['Position'].value))
+    accident_date = parse_date(created)
+    summer_clock = is_nth_weekday(4, 4, accident_date[0], 3)
+    winter_clock = is_nth_weekday(4, 6, accident_date[0], 10)
+
+    # weather is given in UTC time
+    # therefore in summer clock we deduct 3 hours from the local time and in winter clock 2 hours
```

Comment:
can you elaborate? i'm not sure i understand what you mean.

you want me to change this line to date instead of numbers?
 adate = ''.join((accident_date.year, accident_date.month,
accident_date.day, accident_date.hour))

2016-04-17 9:39 GMT+03:00 Daniel Hershcovich <notifications@github.com>:

> In united.py
> <https://github.com/hasadna/anyway/pull/579#discussion_r59979027>:
>
> > +    while node.parentNode:
> > +        node = node.parentNode
> > +        if node.nodeName == "Object":
> > +            return node
> > +
> > +
> > +def all_station_in_date_frame(created):
> > +
> > +    station_data_in_date = collection.getElementsByTagName('date_selected')
> > +    station_data_in_date.sort(key=lambda x: int(x.attributes['Position'].value))
> > +    accident_date = parse_date(created)
> > +    summer_clock = is_nth_weekday(4, 4, accident_date[0], 3)
> > +    winter_clock = is_nth_weekday(4, 6, accident_date[0], 10)
> > +
> > +    # weather is given in UTC time
> > +    # therefore in summer clock we deduct 3 hours from the local time and in winter clock 2 hours
>
> I'd extract our timezone to a constant rather than use plain numbers here.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/hasadna/anyway/pull/579/files/2dd7573bd5bab0315ca68dc10db70ec63fcc8e09#r59979027>
>